packages/web3-eth/src/utils/wait_for_transaction_receipt.ts

```diff
@@ -27,20 +27,23 @@ export async function waitForTransactionReceipt<ReturnFormat extends DataFormat>
 	transactionHash: Bytes,
 	returnFormat: ReturnFormat,
 ): Promise<TransactionReceipt> {
-	return new Promise(resolve => {
+	const pollingInterval =
+		web3Context.transactionReceiptPollingInterval ?? web3Context.transactionPollingInterval;
```

Comment:
can you add a test for this function , by using higher nonce and modified `transactionReceiptPollingInterval` ( to few second ) and look for promise reject after few seconds.

Reply:
Actually raising the nonce revealed 3 issues. And in this MR I am addressing only one of them to make it easier to do the review and merge... and as this issue is somehow separated from others. 
And in a different MR (https://github.com/ChainSafe/web3.js/pull/5311) the nonce is raised in the integration test to validate handling all cases that a provider could respond... :smile:

Reply:
sure you can add test case for this in 5311, good work for finding this issue.